File: Python/CheckDivisibilitybyDigitSumandProduct.py
from functools import reduce

class Solution:
    def checkDivisibility(self, n: int) -> bool:
        digits = list(str(n))

        digitsSum = reduce(lambda acc, cur: acc + int(cur), digits, 0)
        digitsProduct = reduce(lambda acc, cur: acc * int(cur), digits, 1)

        sum = digitsSum + digitsProduct

        if n % sum == 0: return True
        else: return False

        
solution = Solution()


solution.checkDivisibility(99)

File: JavaScript/CheckDivisibilitybyDigitSumandProduct.js
const checkDivisibility = function (n) {
  const digits = n.toString().split("");

  let digitsSum = digits.reduce((acc, cur) => acc + Number(cur), 0);

  let digitsProduct = digits.reduce((acc, cur) => acc * Number(cur));

  let sum = digitsSum + digitsProduct;

  if (n % sum === 0) return true;
  else return false;
};
checkDivisibility(99);
